libsrc/leddevice/dev_spi/LedDeviceSK9822.cpp

```diff
@@ -34,14 +34,15 @@ bool LedDeviceSK9822::init(const QJsonObject &deviceConfig)
 		Info(_log, "[SK9822] Using global brightness control with threshold of %d and max level of %d", _globalBrightnessControlThreshold, _globalBrightnessControlMaxLevel);
 
 		const unsigned int startFrameSize = 4;
-		const unsigned int endFrameSize = qMax<unsigned int>(((_ledCount + 15) / 16), 4);
+		const unsigned int endFrameSize = ((_ledCount/32) + 1)*4;
 		const unsigned int bufferSize = (_ledCount * 4) + startFrameSize + endFrameSize;
 
-		_ledBuffer.resize(bufferSize, 0xFF);
-		_ledBuffer[0] = 0x00;
-		_ledBuffer[1] = 0x00;
-		_ledBuffer[2] = 0x00;
-		_ledBuffer[3] = 0x00;
+		_ledBuffer.resize(0, 0x00);
+		_ledBuffer.resize(bufferSize, 0x00);
```

Comment:
Dont know the sequence of sending frames so that solution works in every condition.
If every frame is initiated before sending then resizing of empty vector to zero is cheap, resize to bufferSize is common cost and fill is unnecessary in that case. If you are sure that's the case then you can remove _ledBuffer.resize(0, 0x00); and not introduce fill.

Reply:
As the `_ledBuffer` vector is shared in `LedDevice.h` and only initialized once, we agree that we have take care filling both the prefix and the suffix with 0x00 bytes for each frame sent to the device.

For most situations this will never be a problem as the vector is initialized with 0x00 and the respective bytes are never set to any other value; unless the length of the ledColors change to a lower value: In this situation filling the suffix with 0x00 is necessary as the position in the vector was previously used for RGB color data.

Research indicates [1] both operations are probably optimized by compiler anyway so I don't see a major pro/con for either solution and your approach seems fine. Thank you for clarification.

[1] https://stackoverflow.com/questions/8848575/fastest-way-to-reset-every-value-of-stdvectorint-to-0

Reply:
if you want go to head with fill then it isnt necessary to zero all the buffer:
std::fill(_ledBuffer.end()-endFrameSize, _ledBuffer.end(), 0x00);
as we need only to take care of frame ending in case if the buffer is shrunk.